src/main/java/uk/gov/hmcts/darts/task/runner/impl/UnstructuredDataExpiryDeletionAutomatedTask.java

```diff
@@ -0,0 +1,165 @@
+
+        deleteTranscriptionDocumentEntity(userAccount, maxRetentionDate, batchSize);
+        deleteMediaEntity(userAccount, maxRetentionDate, batchSize);
+        deleteAnnotationDocumentEntity(userAccount, maxRetentionDate, batchSize);
+        deleteCaseDocumentEntity(userAccount, maxRetentionDate, batchSize);
+    }
+
+    void deleteTranscriptionDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            transcriptionDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredTranscriptionDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getTranscriptionDocumentEntity
+        );
+    }
+
+
+    void deleteMediaEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            mediaRepository,
+            externalObjectDirectoryRepository.findExpiredMediaEntries(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getMedia
+        );
+    }
+
+    void deleteAnnotationDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            annotationDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredAnnotationDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getAnnotationDocumentEntity
+        );
+    }
+
+    void deleteCaseDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            caseDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredCaseDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getCaseDocument
+        );
+    }
+
+
+    <T extends SoftDelete> void deleteExternalObjectDirectoryEntity(
+        UserAccountEntity userAccount,
+        JpaRepository<T, Integer> repository,
+        List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntities,
+        Function<ExternalObjectDirectoryEntity, T> entityMapper) {
+
+
+        List<ExternalObjectDirectoryEntity> deletedEntities = externalObjectDirectoryEntities
+            .stream()
+            .filter(this::deleteFromExternalDataStore)
+            .toList();
+
+        repository.saveAll(
+            deletedEntities
+                .stream()
+                .map(entityMapper)
+                .peek(entity -> entity.markAsDeleted(userAccount))
+                .toList()
+        );
+        externalObjectDirectoryRepository.deleteAll(deletedEntities);
+    }
+
+    boolean deleteFromExternalDataStore(ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
```

Comment:
Ideally the deleters ExternalDataStoreDeleter would know what ExternalLocationTypeEnum applies to them to avoid this if statement

Reply:
This method is used to delete and confirm if deletion was successful so we can filter out unsuccessful deletions :)

The deletion classes will delete anything you provided to it without checking the LocationType the different deletion classes just select the datastore to delete from, as such this is required to ensure we delete from the correct location and only deletes the LocationTypes we want :)